herbieapp/controllers/save_business_entity_controller.py

```diff
@@ -13,15 +13,27 @@
 
 class SaveBusinessEntityController(APIView):
     _entity_manager = None
+    _logger = None
+    _validator = None
+    _schema_registry = None
+    _permission_manager = None
+    _permission_classes = None
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._entity_manager = BusinessEntityManager()
-        self._logger = logging.getLogger(__name__)
-        self._validator = JsonSchemaValidator()
-        self._schema_registry = SchemaRegistry()
+    def __init__(
+            self,
+            _entity_manager: BusinessEntityManager,
+            _logger,
+            _validator: JsonSchemaValidator,
+            _schema_registry: SchemaRegistry,
+            _permission_manager: PermissionManager
```

Comment:
I think the arguments of the init-method don't have to be prefixed with an underscore or is it a convention somehow?

Reply:
Yes this was a bit strange, I think this is a convention for injector. As when I have the variable name in init different as the class variable i get the error
 `received an invalid keyword 'schema_registry'. as_view only accepts arguments that are already attributes of the class`
so as I understand either you have to remove prefix '_' from class variable , in that case we will have self.schema_registry  or prefix must be added also to the injected variable :thinking:

Reply:
hmm okay i was not aware of that. a bit weird to be honest, but ok :+1: